core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -2073,7 +2073,39 @@ class WorkspaceServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matc
     }
   }
 
-"getSubmissionMethodConfiguration" should "return the method configuration that was used to launch the submission" in withTestDataServices { services =>
+  "listWorkspaces" should "log a warning and filter out the workspace if getWorkspace throws an ApiException" in withTestDataServices { services =>
+    val service = services.workspaceService
+    val workspaceId1 = UUID.randomUUID().toString
+    val workspaceId2 = UUID.randomUUID().toString
+
+    // set up test data
+    val azureWorkspace = Workspace("test_namespace1", "name", workspaceId1, new DateTime(), new DateTime(), "testUser1", Map.empty)
+    val googleWorkspace = Workspace("test_namespace2", workspaceId2, workspaceId2, "aBucket", Some("workflow-collection"), new DateTime(), new DateTime(), "testUser2", Map.empty)
+    val googleWorkspaceDetails = WorkspaceDetails.fromWorkspaceAndOptions(googleWorkspace, Some(Set()), true, Some(WorkspaceCloudPlatform.Gcp))
+    val expected = List((googleWorkspaceDetails.workspaceId, googleWorkspaceDetails.cloudPlatform))
+
+    runAndWait {
+      for {
+        _ <- slickDataSource.dataAccess.workspaceQuery.createOrUpdate(azureWorkspace)
+        _ <- slickDataSource.dataAccess.workspaceQuery.createOrUpdate(googleWorkspace)
+      } yield()
+    }
+
+    when (service.workspaceManagerDAO.getWorkspace(azureWorkspace.workspaceIdAsUUID, services.userInfo1.accessToken)).thenAnswer(
```

Comment:
General question:
I'm slightly confused about what is mocked, and what is going to actual services/data sources.
It looks like there's mocking here to handle the WorkspaceDescription, but then the workspaces themselves are created using an actual (connected) slickDataSource?
Why not mock that as well?

I realize that this is consistent with other tests here, and I don't necessarily expect any changes from this: I'm just trying to get context, and maybe provoke a little re-evaluation of our usage patterns.